source/_docs/guides/jira-integration.md

```diff
@@ -0,0 +1,125 @@
+
+In this guide, we will connect a Jira instance to a site on Pantheon. When changes are pushed to Pantheon that reference a jira issue ID, the commit message will appear in the Jira issue's activity log.
+
+
+## What You’ll Need
+
+- A Jira Instance.
+
+- A Drupal or Wordpress site on Pantheon.
+
+- Optional: [Terminus](https://pantheon.io/docs/terminus) installed locally.
+
+## Create a Jira Service Account
+
+1. For security and isolation, on the Jira user administration dashboard, create a user with standard permissions.
+
+![Service account creation](/source/docs/assets/images/integrations/jira/service_account.png)
+
+## Create a Private Folder for Account Credentials.
+
+1. From your local terminal, use the following PHP command to create a JSON file, substituting the below values with your Jira url and the service account's Jira username and password.
+```
+  $> php -r "print json_encode(['jira_url'=>'https://myjira.atlassian.net','jira_user'=>'serviceaccount','jira_pass'=>'secret']);" > secrets.json
+```
+ 
+2. In your Pantheon **files** directory, create a directory called private, if it doesn't exist already. This exact naming convention ensures its contents are not web-accessible. FTP this JSON file to the files/private directory. Repeat this process for all of the sites environments. Each environment should look similar to this:
+
+![Secrets folder](/source/docs/assets/images/integrations/jira/secrets.png)
+
+This upload can also be done quickly with [Terminus](https://pantheon.io/docs/terminus) if it is installed and SFTP commands:
+
+```
+$> `terminus site connection-info --env=dev --site=your-site --field=sftp_command`
+      Connected to appserver.dev.000000-0000-0000-0000-000000000.drush.in.
+  sftp> cd files
+  sftp> mkdir private
+  sftp> cd private
+  sftp> put secrets.json
+```
+
+## Configure Quicksilver Integration
+ 
+1. Clone Pantheon's [Quicksilver Example Repo](https://github.com/pantheon-systems/quicksilver-examples)to your local computer.
+
+```
+git clone https://github.com/pantheon-systems/quicksilver-examples.git quicksilver-examples
+
+```
+
+2. Locate your local copy of your Pantheon site's repository, or clone it if you haven't already.
+
+```
+git clone ssh://codeserver.dev.000000-0000-0000-0000-000000000@codeserver.dev.000000-0000-0000-0000-000000000.drush.in:2222/~/repository.git astro-mktg
+```
+
+3. From the root directory of your local copy (at the same level as index.php), create a directory named "private" and copy jira_integration.php script to it.
+
+```
+$ mkdir private
+$ cd private
+$ cp /path/to/your/quicksilver-examples/jira_integration/jira_integration.php .
+$ ls
+jira_integration.php
+```
+
+4. Add the example jira_integration.php script to the private directory of your code repository.
```

Comment:
`files/private` is on Valhalla, `private` is in the repository. It is, perhaps, unfortunate that these two directories have the same name. The logic, though, is that each storage container (files and repository) have a private location that is not served. Secrets must go in files, not the repository; otherwise, they would not be very secret. Scripts must go in the repository.